ansible/roles/operator-pipeline/templates/openshift/tasks/get-ci-reviewer.yml

```diff
@@ -0,0 +1,100 @@
+
+        # redirecting results variables to /dev/null so they would not be printed
+        echo -n "$list_of_reviewers" | tee $(results.list_reviewers.path) > /dev/null
+        echo -n "$bool_reviewer" | tee $(results.author_is_reviewer.path) > /dev/null
+    - name: approve-pull-request
+      env:
+        - name: GITHUB_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.github_token_secret_name)
+              key: $(params.github_token_secret_key)
+      image: "$(params.pipeline_image)"
+      script: |
+        #! /usr/bin/env bash
+        set -e
+        # do not use -x to not leak github secrets into logs!!!
+        # assigning variables for curl requests
+        PR_NUMBER=$(basename "$(params.pr_url)")
+        REPO_NAME=$(echo "$(params.pr_url)" | sed -e "s|^https://github.com/||" -e "s|/pull/.*$||")
+        API_URL_APPROVE=https://api.github.com/repos/${REPO_NAME}/pulls/${PR_NUMBER}/reviews
+        API_URL_REVIEWERS=https://api.github.com/repos/${REPO_NAME}/pulls/${PR_NUMBER}/requested_reviewers
+        AUTHOR_IS_REVIEWER=$(cat $(results.author_is_reviewer.path))
+        CI_REVIEWERS=$(cat $(results.list_reviewers.path))
+        # requesting review from repo maintainers if there are no reviewers in ci.yaml
+        if [ -z "$CI_REVIEWERS" ]; then
+          curl -L \
+            -X POST \
+            -H "Accept: application/vnd.github.v3+json" \
+            -H "Authorization: bearer ${GITHUB_TOKEN}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            -d '{"reviewers": ["ezopezo", "mporrato", "allda", "tomasbakk", "haripate"]}' \
+            $API_URL_REVIEWERS > /dev/null
+          echo "No reviewers detected in ci.yaml file - requesting review from repo maintainers"
+        else
+         # automatically approving PR if the author of PR is included in list of reviewers
+          if [ "$AUTHOR_IS_REVIEWER" = "true" ]; then
+            curl --X POST \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H "Authorization: bearer ${GITHUB_TOKEN}" \
+              -H "X-GitHub-Api-Version: 2022-11-28" \
+              -d '{"body": "Pull request has been approved", "event": "APPROVE"}' \
+              $API_URL_APPROVE > /dev/null
+            echo "This pull request was automatically approved due to PR author is in the reviewers list"
+          else
+            # request a review from the reviewers listed in ci.yml file if the author is not listed in ci.yaml
+            for reviewer in $CI_REVIEWERS; do
+              curl -L \
+                -X POST \
+                -H "Accept: application/vnd.github.v3+json" \
+                -H "Authorization: bearer ${GITHUB_TOKEN}" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                -d '{"reviewers": ["'"$reviewer"'"]}' \
+                $API_URL_REVIEWERS > /dev/null
+              echo "Requesting review from reviewers in ci.yaml file: $reviewer"
+            done
```

Comment:
Way too much curl sorcery for my taste here :mage:. Can we update the dockerfile for the container image to add the `gh` command line tool package (which is available in the standard fedora repo) and use that instead?
It would make our life a lot easier like:

```sh
gh pr edit $(params.pr_url) --add-reviewer `echo $CI_REVIEWERS | xargs | tr ' ' ','`
gh pr review $(params.pr_url) --approve
```

Reply:
I tried to work with gh package here but it seems that the way how it needs to be authenticated to work with post requests is different from what we already have. I went through the documentation and tried to supply it with token from the env variable as well as calling it with --with-token argument and passing the token this way but I was not able to authenticate the requests. 
But I noticed that we already have github module in operator-pipelines repo https://github.com/redhat-openshift-ecosystem/operator-pipelines/blob/main/operator-pipeline-images/operatorcert/github.py so I can rework the python script to include the post requests there. Then I can get rid of all the curls from bash and also write the tests for the whole thing. WDYT?

Reply:
I updated token's scope, tested it and it works now. Thanks for the inputs guys.